rs/crypto/internal/crypto_lib/threshold_sig/bls12_381/src/ni_dkg/groth20_bls12_381/dealing.rs

```diff
@@ -77,30 +166,23 @@ pub fn create_dealing(
         verify_receiver_indices(receiver_keys, number_of_receivers)?;
     }
 
-    let (public_coefficients, threshold_secret_key_shares) = {
-        if let Some(resharing_secret) = resharing_secret {
-            let resharing_secret: ThresholdSecretKey =
-                ThresholdSecretKey::try_from(&resharing_secret).map_err(|_| {
-                    CspDkgCreateReshareDealingError::MalformedReshareSecretKeyError(
-                        MalformedSecretKeyError {
-                            algorithm: ALGORITHM_ID,
-                            internal_error: "Malformed reshared secret key".to_string(),
-                        },
-                    )
-                })?;
-
-            threshold_share_secret_key(
-                keygen_seed,
-                threshold,
-                number_of_receivers,
-                &resharing_secret,
+    let resharing_secret: ThresholdSecretKey = ThresholdSecretKey::try_from(&resharing_secret)
```

Comment:
nit: The part `: ThresholdSecretKey` is redundant, given that immediately after the `=` it is stated what type this will have.